### PR TITLE
Implement Enhanced Out-of-Distribution (OOD) Detection Metrics and Visualizations

### PR DESCRIPTION
**Pull request by FrozznBrain**
Closes [#24](https://github.com/Cortys/SEP-Probly-WS2526/issues/24)

This PR introduces a unified Out-of-Distribution (OOD) API that acts as a single caller entry point for OOD metrics and visualization utilities, and implements all required and optional OOD evaluation metrics described in the issue.

### Changes
- Added a unified OOD API serving as a central caller for:
  - OOD metric computation
  - OOD visualization utilities
- Implemented required OOD metrics:
  - AUPR (Area Under the Precision-Recall Curve)
  - FPR @ 95% TPR (False Positive Rate at 95% True Positive Rate)
- Implemented optional OOD metrics:
  - FNR @ 95% TPR
  - FPR and FNR at user-defined target rates
- Integrated visualization support into the same unified API:
  - ROC curve plotting
  - In-/Out-of-distribution score visualizations
- Standardized metric interfaces and return formats
- Added comprehensive tests for metrics and unified API behavior

### Motivation
AUROC alone is often insufficient to characterize OOD detection performance.  
Metrics such as AUPR, FPR@95TPR, and FNR@95TPR provide better insight into real-world failure modes.  
The unified API simplifies usage by exposing both metrics and visual diagnostics through a single, consistent interface.

### Testing
- Unit tests added for all required and optional metrics
- Unified API tested as a caller for both metrics and visualizations
- Edge cases (perfect separation, full overlap, degenerate inputs) covered

### Notes
- No breaking changes to the existing public API
- No additional dependencies introduced

### CODE REVIEW

File: src/probly/plot/ood.py
Comment (Cortys) [inline src/probly/plot/ood.py]: Please contact the MEDIX (visualisation) group and discuss whether/how your plotting code could be integrated into our overall visualisation strategy.

File: src/probly/evaluation/types.py
Comment (Cortys) [inline src/probly/evaluation/types.py]: See comment on `visualize_ood`; please remove this type.

File: tests/probly/evaluation/test_tasks.py
Comment (Cortys) [inline tests/probly/evaluation/test_tasks.py]: Since all OOD features are now moved into `ood_api`, the `tasks` tests should not contain any OOD-related tests. Please move those into `test_ood_api`.

File: src/probly/evaluation/ood_api.py
Comment (Cortys) [inline src/probly/evaluation/ood_api.py]: Nice work! You did a good job refactoring the code as requested. To further simplify things, please rename this module to `ood`.

**Review by Cortys — CHANGES_REQUESTED** [review]
Thank you for updating the PR. Overall, this is now very close to a state that we would like to merge. Good job!

I just added some minor comments. Once they are addressed, this PR lgtm.

File: src/probly/evaluation/ood_api.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+        preds = 1.0 - all_scores if invert_scores else all_scores
+
+        # Compute curve arrays conditionally
+        if "roc" in requested_plots:
+            fpr, tpr, _ = roc_curve(labels, preds)
+
+        if "pr" in requested_plots:
+            precision, recall, _ = precision_recall_curve(labels, preds)
+
+    # 3. -> Result Object from types.
```
[the remark this answers is not in the document]
Comment (stacktrce) [reply src/probly/evaluation/ood_api.py]: i did adress all the comments in this PR going to our main (https://github.com/FrozznBrain/probly/pull/39) . Currently we still need to adapt the tests, hence why code checks are failing

**Review by mmschlk — APPROVED** [review]
Looks good from my side, but waiting on final okay from @Cortys. The plot by MEDIX has been added. From my side, I would also be fine not focusing on it now (not removing it now) and doing this later at a next pass.
